aries_cloudagent/messaging/schemas/routes.py

```diff
@@ -115,32 +129,98 @@ async def schemas_send_schema(request: web.BaseRequest):
 
     """
     context: AdminRequestContext = request["context"]
-
-    body = await request.json()
-
-    schema_name = body.get("schema_name")
-    schema_version = body.get("schema_version")
-    attributes = body.get("attributes")
-
-    ledger = context.inject(BaseLedger, required=False)
-    if not ledger:
-        reason = "No ledger available"
-        if not context.settings.get_value("wallet.type"):
-            reason += ": missing wallet-type?"
-        raise web.HTTPForbidden(reason=reason)
-
-    issuer = context.inject(IndyIssuer)
-    async with ledger:
-        try:
-            schema_id, schema_def = await shield(
-                ledger.create_and_send_schema(
-                    issuer, schema_name, schema_version, attributes
+    auto_endorse = json.loads(request.query.get("auto_endorse", "null"))
+
+    if auto_endorse:
+        body = await request.json()
+
+        schema_name = body.get("schema_name")
+        schema_version = body.get("schema_version")
+        attributes = body.get("attributes")
+
+        ledger = context.inject(BaseLedger, required=False)
+        if not ledger:
+            reason = "No ledger available"
+            if not context.settings.get_value("wallet.type"):
+                reason += ": missing wallet-type?"
+            raise web.HTTPForbidden(reason=reason)
+
+        issuer = context.inject(IndyIssuer)
+        async with ledger:
+            try:
+                schema_id, schema_def = await shield(
+                    ledger.create_and_send_schema(
+                        issuer, schema_name, schema_version, attributes
+                    )
                 )
+            except (IndyIssuerError, LedgerError) as err:
+                raise web.HTTPBadRequest(reason=err.roll_up) from err
+
+        return web.json_response({"schema_id": schema_id, "schema": schema_def})
+
+    else:
+        transaction_message = await request.json()
+        session = await context.session()
+
+        wallet: BaseWallet = session.inject(BaseWallet, required=False)
```

Comment:
This code is all duplicated from existing code in the ledger object, is there some way to reuse (e.g. call ledger to create the transaction object) rather than repeating the code?